qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -200,6 +202,24 @@ def _t_complexity_(self) -> 'TComplexity':
         ((toffoli, n),) = self.bloq_counts().items()
         return n * toffoli.t_complexity()
 
+    def add_my_tensors(
+        self,
+        tn: 'qtn.TensorNetwork',
+        tag: Any,
+        *,
+        incoming: Dict[str, 'SoquetT'],
+        outgoing: Dict[str, 'SoquetT'],
+    ):
+        import quimb.tensor as qtn
+
+        N, M = 2**self.x_bitsize, 2**self.phase_bitsize
+        inds = (incoming['x'], incoming['phase_grad'], outgoing['x'], outgoing['phase_grad'])
+        unitary = np.zeros((N, M) * 2, dtype=np.complex128)
+        for a, b in itertools.product(range(N), range(M)):
+            unitary[a, b, a, int(math.fmod(a + b, M))] = 1
+
+        tn.add(qtn.Tensor(data=unitary, inds=inds, tags=[self.short_name(), tag]))
+
```

Comment:
The `add_my_tensorts` would be a bit more complex now given the new parameters `sign` and `right_shift`. Since we already have a unitary implemented, we can derive tensor data via

```suggestion
    def add_my_tensors(
        self,
        tn: 'qtn.TensorNetwork',
        tag: Any,
        *,
        incoming: Dict[str, 'SoquetT'],
        outgoing: Dict[str, 'SoquetT'],
    ):
        from qualtran.cirq_interop._cirq_to_bloq import _add_my_tensors_from_gate

        _add_my_tensors_from_gate(
            self, self.signature, self.short_name(), tn, tag, incoming=incoming, outgoing=outgoing
        )
```